File: Bedu/Java/Clase3/SharePreFrame/app/src/main/java/com/charlye934/sharepreferencestutorial/MainActivity.java
package com.charlye934.sharepreferencestutorial;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    Button btnSig,btnGuardar;
    ImageView img1, img2, img3;
    EditText txtNombre, txtPass;
    LinearLayout linearLayout;
    SharedPreferences sharePref;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        sharePref = getApplicationContext().getSharedPreferences("archivo", Context.MODE_PRIVATE);

        if(leerPreferencias()) {
            Toast.makeText(getApplicationContext(), "Ya iniciado", Toast.LENGTH_SHORT).show();
            cambiarPagina();
        }else{

            img1 = (ImageView) findViewById(R.id.img1);
            img2 = (ImageView) findViewById(R.id.img2);
            img3 = (ImageView) findViewById(R.id.img3);
            txtNombre = (EditText) findViewById(R.id.txtNombre);
            txtPass = (EditText) findViewById(R.id.txtPass);
            linearLayout = (LinearLayout) findViewById(R.id.linerLayout);
            btnSig = (Button) findViewById(R.id.btnSig);
            btnGuardar = (Button) findViewById(R.id.btnGuardar);

            btnSig.setOnClickListener(
                    new View.OnClickListener() {
                        @Override
                        public void onClick(View v) {
                            opcSig();
                        }
                    }
            );

            btnGuardar.setOnClickListener(
                    new View.OnClickListener() {
                        @Override
                        public void onClick(View v) {
                            guardar();
                        }
                    }
            );
        }
    }

    void opcSig(){
        if(img1.getVisibility() == View.VISIBLE) {
            img1.setVisibility(View.INVISIBLE);
            img2.setVisibility(View.VISIBLE);
        }else if(img2.getVisibility() == View.VISIBLE){
            img2.setVisibility(View.INVISIBLE);
            img3.setVisibility(View.VISIBLE);
        }else if(img3.getVisibility() == View.VISIBLE){
            img3.setVisibility(View.INVISIBLE);
            btnSig.setVisibility(View.INVISIBLE);
            linearLayout.setVisibility(View.VISIBLE);
        }
    }

    void guardar(){
        String nombre = txtNombre.getText().toString();
        String pass = txtPass.getText().toString();

        if(nombre.length() != 0 && pass.length() != 0){
            SharedPreferences.Editor obj_editor = sharePref.edit();
            obj_editor.putString("nombre", nombre);
            obj_editor.putString("pass", pass);
            obj_editor.commit();
            Toast.makeText(getApplicationContext(),"Se guardaron correctamente los datos", Toast.LENGTH_SHORT).show();
            cambiarPagina();
        }else{
            Toast.makeText(getApplicationContext(),"Se deben de llenar los campos", Toast.LENGTH_SHORT).show();
        }
    }

    Boolean leerPreferencias(){
        String nombres = sharePref.getString("nombre",null);
        return nombres != null;
    }

    void cambiarPagina(){
        Intent intent = new Intent(getApplicationContext(),Main2Activity.class);
        startActivity(intent);
    }

}

File: Bedu/Kotlin/Clase1-3/SharePreFrame/app/src/main/java/com/charlye934/sharepreferencestutorialkotlin/MainActivity.kt
package com.charlye934.sharepreferencestutorialkotlin

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val sharePreferences:SharedPreferences = applicationContext.getSharedPreferences("archivo", Context.MODE_PRIVATE)
        if (leerPreferencias(sharePreferences)) {
            Toast.makeText(applicationContext, "Ya iniciado", Toast.LENGTH_LONG).show()
            cambiarPagina()
        } else {
            btnSig.setOnClickListener{opcSig()}
            btnGuardar.setOnClickListener { guardar(sharePreferences) }
        }
    }

    fun leerPreferencias(sharedPreferences: SharedPreferences ): Boolean {
        var nombres = sharedPreferences.getString("nombre", null)
        return nombres != null
    }

    fun opcSig(){
        if(img1.visibility == View.VISIBLE){
            img1.setVisibility(View.INVISIBLE)
            img2.setVisibility(View.VISIBLE)
        }else if(img2.visibility == View.VISIBLE){
            img2.setVisibility(View.INVISIBLE)
            img3.setVisibility(View.VISIBLE)
        }else if(img3.visibility == View.VISIBLE){
            img3.visibility = View.INVISIBLE
            btnSig.visibility = View.INVISIBLE
            linerLayout.setVisibility(View.VISIBLE)
        }
    }

    fun guardar(sharedPreferences: SharedPreferences){
        var nombre = txtNombre.text.toString()
        var pass = txtPass.text.toString()
        if(!nombre.isEmpty() && !pass.isEmpty()){
            var objEditor = sharedPreferences.edit()
            objEditor.putString("nombre", nombre)
            objEditor.commit()
            Toast.makeText(applicationContext, "Se guardaron correctamente los datos", Toast.LENGTH_SHORT).show()
            cambiarPagina()
        }else{
            Toast.makeText(applicationContext, "Se deben de llenar los campos", Toast.LENGTH_SHORT).show()
        }
    }

    fun cambiarPagina() {
        val intent = Intent(this, Main2Activity::class.java)
        startActivity(intent)
    }
}
